### PR TITLE
Daemon http paths and params

### PR DESCRIPTION
**Pull request by valmack**
### Motivation

We want to be able to show requests per endpoint in our monitoring dashboards

### Changes

- Use express routers to organize request url, path, and params
- Add params to http-access logs
- Clean up http logs with string wrapping (to follow [logfmt](https://brandur.org/logfmt))

### Notes

1. It's harder to read the endpoints now in the code. Generally these routes would be organized in separate files. I'm down to move towards making this cleaner and more organized if people agree it would be helpful. Quick change.

2. 404s will not have these log properties assigned properly because they don't lead to any real handler. So the logs will look like the old format.

Old format
```js
[Thu, 06 May 2021 19:35:31 GMT]
service=http-access
ip=35.203.252.113
ts=2021-05-06T19:35:31.668Z
method=GET
// notice urls and path
original_url=/api/v0/documents/k2t6wyfsu4pg1bu91pkuskd2ig92gldvd3hz6sy9rbuk8zmp2g3fds5ju7j6l1
base_url=-
path=/api/v0/documents/k2t6wyfsu4pg1bu91pkuskd2ig92gldvd3hz6sy9rbuk8zmp2g3fds5ju7j6l1
http_version=1.1
req_header-
status=200
content_length=370
content_type="application/json; charset=utf-8"
ref=-
user_agent=node-fetch/1.0 (+https://github.com/bitinn/node-fetch)
elapsed_ms=1106.483
error_message=-
error_code=-
```

New format
```js
[Thu, 06 May 2021 19:41:10 GMT]
service=http-access
ip=127.0.0.1
ts=2021-05-06T19:41:10.211Z
method=GET
// notice urls, path, params
original_url=/api/v0/documents/k2t6wyfsu4pg2r88kogtioocim6p6ewxnhakzau0ekd4xmparhncekviu7nvuq/
base_url=/api/v0/documents
path=/k2t6wyfsu4pg2r88kogtioocim6p6ewxnhakzau0ekd4xmparhncekviu7nvuq/
params.docid=k2t6wyfsu4pg2r88kogtioocim6p6ewxnhakzau0ekd4xmparhncekviu7nvuq
http_version=1.1
req_header-
status=200
content_length=354
content_type="application/json; charset=utf-8"
ref=-
user_agent="insomnia/2021.2.2"
elapsed_ms=4.742
error_message="-"
error_code=-
```

### CODE REVIEW

File: packages/cli/src/ceramic-daemon.ts [hunk 1]
```diff
@@ -180,34 +179,53 @@ class CeramicDaemon {
...
     }
+
+    baseRouter.use('/commits', commitsRouter)
+    baseRouter.use('/documents', documentsRouter)
+    baseRouter.use('/multiqueries', multiqueriesRouter)
+    baseRouter.use('/node', nodeRouter)
+    baseRouter.use('/pins', pinsRouter)
+    baseRouter.use('/records', recordsRouter)
+    baseRouter.use('/streams', streamsRouter)
+    baseRouter.use(errorHandler(this.diagnosticsLogger))
```
Comment (ukstv) [inline packages/cli/src/ceramic-daemon.ts:226]: @valmack This line should not be necessary.

**Review by ukstv — APPROVED** [review]

File: packages/cli/src/ceramic-daemon.ts (hunk 1 above)
Comment (valmack) [reply packages/cli/src/ceramic-daemon.ts:226]: I guess you are saying line 226 with the errorHandler?

But it actually needs to be added to each individual router to properly parse params as well. If we don't use it in these routers our "requests per endpoint" will exclude failed requests. So I'm actually going to add it in more places.

With
```
original_url=/api/v0/documents/k2t6wyfsu4pg2r88kogtioocim6p6ewxnhakzau0ekd4xmparhncekviu7nvua/
base_url=/api/v0/documents
path=/k2t6wyfsu4pg2r88kogtioocim6p6ewxnhakzau0ekd4xmparhncekviu7nvua/
params=-
```

Without
```
original_url=/api/v0/documents/k2t6wyfsu4pg2r88kogtioocim6p6ewxnhakzau0ekd4xmparhncekviu7nvua/
base_url=/api/v0
path=/documents/k2t6wyfsu4pg2r88kogtioocim6p6ewxnhakzau0ekd4xmparhncekviu7nvua/
params=-
```

Comment (ukstv) [reply packages/cli/src/ceramic-daemon.ts:234]: Wow. That's sad and concerning.

File: packages/cli/src/ceramic-daemon.ts [hunk 2]
```diff
@@ -180,34 +179,61 @@ class CeramicDaemon {
...
+
+    commitsRouter.use(errorHandler(this.diagnosticsLogger))
+    documentsRouter.use(errorHandler(this.diagnosticsLogger))
+    multiqueriesRouter.use(errorHandler(this.diagnosticsLogger))
+    nodeRouter.use(errorHandler(this.diagnosticsLogger))
+    pinsRouter.use(errorHandler(this.diagnosticsLogger))
+    recordsRouter.use(errorHandler(this.diagnosticsLogger))
+    streamsRouter.use(errorHandler(this.diagnosticsLogger))
+
+    baseRouter.use('/commits', commitsRouter)
```
Comment (stbrody) [inline packages/cli/src/ceramic-daemon.ts:227]: Nitpick: Can we put this block up above the bindings to specific endpoints so it's easier while reading through this `registerAPIPaths` function to follow the flow and figure out what the various API endpoints are?

**Review by stbrody — APPROVED** [review]

**Pull request merged**